From 51fbca93e8bab5bb68ae4cd77ffa61ac3e8d2f08 Mon Sep 17 00:00:00 2001
From: Jerry Cao <jerry.cao@amlogic.com>
Date: Fri, 9 May 2014 14:47:52 -0700
Subject: [PATCH 4041/5965] Reset M6 kernel task CPU affinity to all CPU after
 invoking TrustZone EFUSE API.

After TrustZone EFUSE API (used when TEE is running) is invoked, it does not
reset the kernel task's CPU affinity mask back to all CPU upon completion.

This results in the kernel task can only be executed on CPU0 and degrades
overall system performance.
---
 arch/arm/mach-meson6/meson-secure.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-meson6/meson-secure.c b/arch/arm/mach-meson6/meson-secure.c
index 3ea6dc27568d..cd01d21d7b0f 100755
--- a/arch/arm/mach-meson6/meson-secure.c
+++ b/arch/arm/mach-meson6/meson-secure.c
@@ -143,6 +143,7 @@ int meson_trustzone_efuse(struct efuse_hal_api_arg* arg)
 	}
 	outer_inv_range((arg->retcnt_phy), (arg->retcnt_phy + sizeof(unsigned int)));
 	dmac_unmap_area(__va(arg->buffer_phy), arg->size, DMA_FROM_DEVICE);
+	set_cpus_allowed_ptr(current, cpu_all_mask);
 
 	return ret;
 }
-- 
2.19.0

